### ISSUE TITLE
D in visual mode does not yank full lines

### ISSUE THREAD

**Issue by sheam** [sublimehq/Vintage#132]
```
line one
line two
```

putting cursor on the e on line 1, then pressing v (vis mode), then j (to highlight next line), then D should yank both lines completely. Instead I am left with 

```
lin
```

In actual vi or vim, all of line 1 and 2 would be yanked.

**Comment by sheam** [comment]
In other words, vjD should be exactly the same as 2dd, or vjjD should behave exactly the same as 3dd.

**Comment by owengalenjones** [comment]
This doesn't appear to be the case for Y so with the same example:

```
lin|e one
line two
```

putting cursor on the e on line 1, then pressing v (vis mode), then j (to highlight next line), then Y then jp results in:

```
line one
line two
line one
line two
```

**Comment by sheam** [comment]
interesting, I would have assumed Y would have been the same.

**Comment by owengalenjones** [comment]
However C works just like D so in the example

```
lin|e one
line two
```

where the '|' is the cursor hitting C will result in

```
lin|
```

and in insert mode.

**Comment by owengalenjones** [comment]
Hey sheam, try my pull request to see if it does everything as expected from vi/vim.

**Comment by sheam** [comment]
I can give it a try right now. Do I just download the file you changed, and save it into \Sublime Text 2\Data\Packages\Default\Default.sublime-keymap? At work here I am on windows, using ST2 portable.

**Comment by sheam** [comment]
Ok, figured it out. Downloaded it to my Packages\Vintage folder. 

It does work much better now. Well I guess it is 100% correct in how it yanks.

But it  is still affect when you paste at the end of the file: if you paste on last line, it won't paste underneath last line, but instead at end of last line. But this is likely an entirely different bug.

So I vote (not that I get one) to Owen's fix and merge it into trunk.

**Comment by owengalenjones** [comment]
I closed my pull request, there was an issue with how the selection is expanded and visual mode.  

Example: ( '`|`' is the cursor )

```
Looooooooong li|ne  
Short line*
Other line
```

If you start visual mode in the 'Long Line', go down to to 'Short Line' and hit D, or C it will also delete 'Other Line.'  This bug exists for Y as well and is due to ( I think ) how expand_to_full_line works in conjunction with visual mode ie. in visual mode when you hit j to go down to short line the cursor will be a space after the end of 'Short Line' (where the \* is).  Then when you run the expand selection to line command you can see it will also select 'Other Line'

I will look into how to get this to run, I really want to use ST2 but perfect Vintage mode is a must for me :(

**Comment by sheam** [comment]
I agree.  I wonder if they are going to be able to accomplish it with just a series of keybindings.  I really think it needs to be thought of from day one of development.

**Comment by bobbydavid** [comment]
I wish I could upvote this vjjD != 2D issue. After a few weeks of using Sublime, this issue may force me back to Vim. Is anyone in the position to work on this? I really want to use ST2 but I feel like I'm making a double-or-nothing investment to learn to fix this bug on top of learning ST2....
